### PR TITLE
Handle blocked copilot and Expensify card flows gracefully

### PR DESCRIPTION
**Pull request by ChavdaSachin**
### Explanation of Change

This PR aims to restrict copilots from performing certain actions on behalf of the owner.
As A Part of Solution Primarily Show DelegetNoAccessModal when copilot tries to perform any restricted action
And restrict delegate from accessing any restricted page using deeplink by wrapping those pages with DelegateNoAccessWrapper.

List of Actions Restricted for Copilot on Behalf of Owner,
- Modifying Private Profile Credentials(Legal Name, DOB, Phone Number, Address).
- Add, Delete or Modify Bank Account.
- Enable Wallet.
- Enable/Disable 2FA.
- Close Account.
- Add, Remove or Change Access Level of other Copilots.
- Modify Subscription Plans/ Add Payment Card for Subscription.
- Switch To Expensify Classic.
- Issue New Expensify Card,Add New Company Card, Assign Company Card

### Fixed Issues

$ https://github.com/Expensify/App/issues/50796
PROPOSAL: NA

### Tests

Same As QA.
- [x] Verify that no errors appear in the JS console

### Offline tests

Same As QA.

### QA Steps

>[!IMPORTANT]
>All of the following tests should be performed as a copilot.
>For Steps To Add Copilot Visit - https://help.expensify.com/articles/expensify-classic/copilots-and-delegates/Assign-or-remove-a-Copilot

Test 1.
1. Go to Settings > Profile.
2. Press Legal Name.
3. Verify "Not So Fast..." Modal pops up when trying to save.
4. Perform Step 2-3 for Date of Birth, Phone Number, Address.

Test 2.
1. Go to Settings > Wallet.
2. Try to add/remove Bank Account.
5. Verify "Not So Fast..." Modal Pops Up When Trying to add/remove Bank Account.

Test 3.(For Accounts With Wallet Not Enabled)
1. Go to Settings > Wallet.
2. Click Enable Wallet.
3. Verify "Not So Fast..." Modal Pops Up.

Test 4.
1. Go to Settings > Security.
2. Click Two Factor Authentication/Close Account.
3. Verify "Not So Fast..." Modal Pops Up.

Test 5.
1. Go to Settings > Security.
2. Open ThreeDot Menu for Any Copilot.
3. Click Change Access Level/Remove Copilot.
6. Verify "Not So Fast..." Modal Pops Up.

Test 6.
1. Go to Settings > Subscription.
2. Click Add Payment Card.
3. Verify "Not So Fast..." Modal Pops Up.
5. Repeat step 2-3 for Request Early Cancellation, Change Plan(Annual<->Pay Per Use),Subscription size option(Only visible if current plan is annual), Toggle Auto Renew, Toggle Auto-Increase annual seats.

Test 7.
1. Go to Settings.
2. Click Switch to Expensify Classic.
3. Verify "Not So Fast..." Modal Pops Up.

Test 8.
1. Visit Deeplink https://dev.new.expensify.com:8082/settings/profile/contact-methods/new.(Note - change domain as per the need ie. `staging.newExpensify.com` or `new.expensify.com` from `dev.new.expensify.com:8082`).
2. Verify No Access Page is Visible.
3. Repeat step 1-2 for following Deeplinks.
4. 

  - https://dev.new.expensify.com:8082/settings/profile/legal-name
  - https://dev.new.expensify.com:8082/settings/profile/date-of-birth
  - https://dev.new.expensify.com:8082/settings/profile/phone
  - https://dev.new.expensify.com:8082/settings/profile/address
  - https://dev.new.expensify.com:8082/settings/wallet/add-bank-account
  - https://dev.new.expensify.com:8082/settings/wallet/enable-payments
  - https://dev.new.expensify.com:8082/settings/security/two-factor-auth
  - https://dev.new.expensify.com:8082/settings/security/closeAccount
  - https://dev.new.expensify.com:8082/settings/security/delegate
  - https://dev.new.expensify.com:8082/settings/subscription/add-payment-card
  - https://dev.new.expensify.com:8082/settings/subscription/request-early-cancellation-survey
  - https://dev.new.expensify.com:8082/settings/subscription/subscription-size
  - https://dev.new.expensify.com:8082/settings/exit-survey/book-call
  - https://dev.new.expensify.com:8082/settings/exit-survey/reason?backTo=settings%2Fexit-survey%2Fbook-call
  - https://dev.new.expensify.com:8082/settings/exit-survey/response?reason=featureNotAvailable&backTo=settings%2Fexit-survey%2Freason
  - https://dev.new.expensify.com:8082/settings/exit-survey/response?reason=dontUnderstand&backTo=settings%2Fexit-survey%2Freason
  - https://dev.new.expensify.com:8082/settings/exit-survey/response?reason=preferClassic&backTo=settings%2Fexit-survey%2Freason
  - https://dev.new.expensify.com:8082/settings/exit-survey/confirm?backTo=settings%2Fexit-survey%2Fresponse%3Freason%3DpreferClassic%26backTo%3Dsettings%252Fexit-survey%252Freason
  - https://dev.new.expensify.com:8082/settings/exit-survey/confirm?backTo=settings%2Fexit-survey%2Fresponse%3Freason%3DdontUnderstand%26backTo%3Dsettings%252Fexit-survey%252Freason
  - https://dev.new.expensify.com:8082/settings/exit-survey/confirm?backTo=settings%2Fexit-survey%2Fresponse%3Freason%3DfeatureNotAvailable%26backTo%3Dsettings%252Fexit-survey%252Freason
  
  Please replace a valid workspaceID in following list while testing.
  - https://dev.new.expensify.com:8082/settings/workspaces/{workspaceID}/expensify-card/issue-new 
  - https://dev.new.expensify.com:8082/settings/workspaces/{WorkspaceID}/expensify-card/choose-bank-account
  - https://dev.new.expensify.com:8082/settings/workspaces/{workspaceID}/company-cards/add-card-feed
  
Test 9.
1. Go to Settings > Workspaces.
2. Open one of the workspace with no Expensify Cards allotted.
3. Go to Expensify Cards Section.
4. Click `Issue New Card` Button.
5.  Verify "Not So Fast..." Modal Pops Up.
6. Go to Settings > Workspaces.
7. Open a workspace with one or more Expensify Cards allotted.
8. Go to Expensify Cards Section.
9. Click `+ Issue Card` Button.
10.  Verify "Not So Fast..." Modal Pops Up.

Test 10.
1. Go to Settings > Workspaces.
2. Open one of the workspace with no Company Cards allotted.
3. Go to Company Cards Section.
4. Click `Add Cards` Button.
5.  Verify "Not So Fast..." Modal Pops Up.
6. Go to Settings > Workspaces.
7. Open a workspace with one or more Company Cards allotted.
8. Go to Company Cards Section.
9. Select a Feed if Needed.
10. Click `+ Issue Card` Button.
11.  Verify "Not So Fast..." Modal Pops Up.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

Test 1-7

https://github.com/user-attachments/assets/f7aba85d-1ca9-4556-9be6-9eb2de97e90b

Test 8

https://github.com/user-attachments/assets/e0e28b02-760c-48e5-96dd-e2263562c545

</details>

<details>
<summary>Android: mWeb Chrome</summary>

Test 1-7

https://github.com/user-attachments/assets/9d6db304-bce4-41ad-81ac-02b853bc7bb4

Test 8

https://github.com/user-attachments/assets/63ac4329-8184-4f15-a488-93ce2fbc16a5

</details>

<details>
<summary>iOS: Native</summary>

Test 1-7

https://github.com/user-attachments/assets/c853766b-3555-4566-940f-ef883acbf17e

Test 8

https://github.com/user-attachments/assets/8caea800-404a-423b-8c4e-e284c1129bc9

</details>

<details>
<summary>iOS: mWeb Safari</summary>

Test 1-7

https://github.com/user-attachments/assets/71579adb-0bdd-48ab-8bf7-309d79599241

Test 8

https://github.com/user-attachments/assets/39037985-e479-4d6e-8465-05586eed89cf

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

Test 1-7

https://github.com/user-attachments/assets/14f53224-69ff-4e03-a216-1f1eaf6a4fb0

Test 8

https://github.com/user-attachments/assets/0e0e046c-f1cf-4b89-879c-db310316490e

</details>

<details>
<summary>MacOS: Desktop</summary>

Test 1-7

https://github.com/user-attachments/assets/22005d14-84fa-4527-91df-74a95fb1401e

Test 8

https://github.com/user-attachments/assets/b8e54110-4d67-430c-a73e-06c5e911e28b

</details>

### CODE REVIEW

**Comment by ChavdaSachin** [comment]
@Julesssss draft is up.
Working on DelegateNoAccessWrapper to accept PageNotFound props, and subscription page.

Quick question -
On security page in delegates section, both menu items are restricted actions, should we hide three dot menu there?
Screenshot 2024-11-06 at 5 18 39 PM

Same question for payment methods list on wallet page.

**Comment by ChavdaSachin** [comment]
One more important note here,

For privateProfileOptions fields (legal name,legal address......) following two options came to my mind.
1. Show noDelegateAccessModal as soon as delegate clicks on option from profile page.
2. Let delegate open RHP edit page for the option but show noDelegateAccessModal as soon as delegate tries to submit.

IMO option 2 aligns more with rest of the app.

Now again for option 2 there was an important decision about validation.
- I have decided to skip validation check for the actions where delegate would ultimately see noDelegateAccessModal when trying to submit.
- Or delegate might take time to clear all validations and at the end would not be able to make any changes, that is kind of frustrating for the user.

Thoughts?

**Review by Julesssss — COMMENTED** [review]
We have [design approval](https://github.com/Expensify/App/issues/50796#issuecomment-2466991729) now, thanks for sharing the questions!

**Comment by ChavdaSachin** [comment]
One more task for design team,
When a delegate use deeplink to access the page which is restricted what message should we display?
I suggest the message could be "As a delegate you don't have access to this page." with title "Hmm... it's not here"....
Screenshot 2024-11-14 at 3 42 29 PM

Sounds good?

**Comment by shawnborton** [comment]
I think something like that would work, but I could also see just using the same exact "Not found" page we use elsewhere in terms of the exact copy and text. cc @Expensify/design

**Comment by ChavdaSachin** [comment]
Currently we display either of "Oops, this page cannot be found" or "You don't have access to this chat" for not found page subtitle.

**Comment by shawnborton** [comment]
Got it, so maybe something like:

Title: You don't have access to this
Body: As a delegate, you can not access this page.

Or something like that. cc @jamesdeanexpensify for a quick looksie.

**Comment by ChavdaSachin** [comment]
One more input request,
Which of the following pages should contain back button for page not found view?

1. Desktop - Full screen page not found view.
2. Desktop - RHP page not found.
3. Mobile - page not found view.

**Comment by shawnborton** [comment]
Hmm I would think all of the, right? How do the current "not found" pages work?

**Comment by ChavdaSachin** [comment]
I checked, and yes back button is visible on all three case.
It is hidden for the only case where LHP is visible along with not found view on right side...
So this sounds resolved.
Screenshot 2024-11-14 at 4 14 10 PM

**Comment by dannymcclain** [comment]
Definitely agree with going with copy something like this (Shawn's suggestion):

>**You don't have access to this**
>As a delegate, you can not access this page.

I think there's a pretty decent difference between "We couldn't find that page" and "You don't have access to this". 
The first message would make me think the product is broken or someone sent me a bum link, the second one is super clear about what's actually happening.

**Comment by jamesdeanexpensify** [comment]
For the [copy](https://github.com/Expensify/App/pull/52103#issuecomment-2475987121), what about:

Title: Not so fast... 
Body: As a copilot, you don't have access to this page. Sorry!

**Question though** - are we using this copy/modal for another type of "delegate" as well (like vacation delegate), or just for the copilot scenario? I'm just seeing both terms so I wanted to be sure what we should use in the modal.

**Comment by jamesdeanexpensify** [comment]
Cool! I know we're starting to build "vacation delegate" functionality, but maybe we don't have to worry about that here. So we can go with [this](https://github.com/Expensify/App/pull/52103#issuecomment-2477060795)!

**Comment by ChavdaSachin** [comment]
So if I am interpreting it correct,

@jamesdeanexpensify is suggesting:
>Title: Not so fast...
>Body: As a copilot, you don't have access to this page. Sorry!

While @shawnborton suggested:
>Title: You don't have access to this
>Body: As a ~delegate~Copilot, you can not access this page.

Waiting for final decision from design team.

**Comment by shawnborton** [comment]
Generally we're happy to defer to @jamesdeanexpensify in these situations :) So I say feel free to go with his copy unless anyone on the D-team feels strongly otherwise.

**Comment by ChavdaSachin** [comment]
https://github.com/Expensify/App/pull/52103#issuecomment-2459596310
> * I have decided to skip validation check for the actions where delegate would ultimately see noDelegateAccessModal when trying to submit.

Please keep this important detail in mind while reviewing.

**Comment by ChavdaSachin** [comment]
@Julesssss manually tagging you here, coz Malvin again tagged Jon here.

File: src/pages/settings/Profile/Contacts/NewContactMethodPage.tsx
Comment (c3024) [inline src/pages/settings/Profile/Contacts/NewContactMethodPage.tsx:111]: `DelegateNoAccessWrapper` is wrapped within `ScreenWrapper` everywhere else except here. Any reason for this?

**Review by c3024 — COMMENTED** [review]
Great job. 🙇 

I will test and see if I find anything.

Couple of comments.

**Comment by c3024** [comment]
@ChavdaSachin, you raised [this](https://github.com/Expensify/App/pull/52103#issuecomment-2459596310), but I somehow missed it—sorry!

The first option seems better to me.

Currently, with the changes in the PR, the delegate goes to the legal name (or date of birth or address) page, fills out the fields, clicks on Submit, and then discovers they cannot edit them. Furthermore, they need to dismiss the “no access” modal and then the RHP to navigate to the profile details page.

https://github.com/user-attachments/assets/8c8d77c7-d807-492e-ace2-05bfd5702118

cc: @Expensify/design

File: src/pages/settings/Profile/Contacts/NewContactMethodPage.tsx
Comment (ChavdaSachin) [reply src/pages/settings/Profile/Contacts/NewContactMethodPage.tsx:111]: It is a mistake, It should be inside `ScreenWrapper`. Fixing...

**Comment by ChavdaSachin** [comment]
@c3024 I thought option 2 was more appropriate coz that is the exact same flow we implemented for add new contact method in [This PR](https://github.com/Expensify/App/pull/50643), based on [This Comment](https://github.com/Expensify/App/issues/50063#issuecomment-2399975948) from design team.

> Currently, with the changes in the PR, the delegate goes to the legal name (or date of birth or address) page, fills out the fields, clicks on Submit, and then discovers they cannot edit them. Furthermore, they need to dismiss the “no access” modal and then the RHP to navigate to the profile details page.

Regarding this hideous flow you mentioned, We could dismiss RHP as soon as user close "Not So Fast..." Modal.

https://github.com/user-attachments/assets/22c0605c-3155-47ec-811f-1b952f5f9033


https://github.com/user-attachments/assets/02585478-70ec-4625-bea2-8d1e0e5e8735

**Comment by dannymcclain** [comment]
Personally I think it would be better to pop open the not so fast modal as soon as the user clicked on the option row. I think it's weird to let them edit the value and then only show the modal when they try to save.

I also think that's inline with what we did for contact method, because in that flow we're not letting the user edit anything—we're popping the modal as soon as they click something that _would_ allow them to edit.

>So, I think showing the “no access” modal directly when the option is clicked and displaying the [“not found” page](https://github.com/Expensify/App/pull/52103#issuecomment-2485358638) when deep-linked to these pages, like you did for other pages, is better.

Basically I agree with the above comment ☝️

**Comment by ChavdaSachin** [comment]
Great, this might work

**Comment by Julesssss** [comment]
This is looking good. Seems like we just have some lint issues remaining.

**Comment by ChavdaSachin** [comment]
a small change and PR will be ready for review

**Comment by ChavdaSachin** [comment]
Ready for review

**Comment by c3024** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.ts or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] For any bug fix or new feature in this PR, I verified that sufficient [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) are included to prevent regressions in this flow.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by c3024 — APPROVED** [review]
LGTM!

**Comment by jamesdeanexpensify** [comment]
Sorry if I'm late here, but I wonder if we should align [this modal](https://github.com/Expensify/App/pull/52103#issuecomment-2485302863) with the copy suggestion for [this page](https://github.com/Expensify/App/pull/52103#issuecomment-2485358638) more, for consistency's sake. What do you all think? So then it might be:

**Page**
Title: Not so fast...
Body: As a copilot, you don't have access to this page. Sorry!

**Modal**
Title: Not so fast...
Body: As a copilot, you don't have permission to take this action. Sorry!

**Comment by jamesdeanexpensify** [comment]
It's probably fine not to include {accountOwnerEmail}, but we could also do:

**Page**
Title: Not so fast...
Body: As a copilot for {accountOwnerEmail}, you don't have access to this page. Sorry!

**Modal**
Title: Not so fast...
Body: As a copilot for {accountOwnerEmail}, you don't have permission to take this action. Sorry!

**Comment by ChavdaSachin** [comment]
Proceeding with this copy, as latest suggestion/agreement from design team.

> **Modal**
> Title: Not so fast...
> Body: As a copilot for {accountOwnerEmail}, you don't have permission to take this action. Sorry!

**Comment by ChavdaSachin** [comment]
SS of the latest copy

Screenshot 2024-12-03 at 3 24 09 PM
Screenshot 2024-12-03 at 3 25 54 PM

**Comment by ChavdaSachin** [comment]
Hey @c3024, we made a blunder here.
While completing [This List](https://github.com/Expensify/App/issues/50796#issuecomment-2419213330) I completely missed the original issue here, and forgot to implement the same for Expensify Cards and Company Cards.

https://github.com/Expensify/App/issues/50796#issue-2588721682
> We should implement exact same solution for issue new Expensify card flow, additionally we need to check for company cards as well.

Just now I made those changes, and I need your help for testing. Could you please provide mock data/mock Onyx data for company card feeds so I could test those(I am not yet invited to Slack so I don't have access to any sandbox data).
Thank you.

**Comment by c3024** [comment]
I have not found any sandbox data for cards, and AFAIK, no such data is available for cards.

Since the Copilot flow is restricted on the initial clicks on most pages, such as issuing pages, we can test these flows without adding any cards. We can include tests for flows that require card data in QA tests. In the unlikely event that QA finds a bug for these flows, they will raise an issue.

cc: @Julesssss

**Comment by ChavdaSachin** [comment]
Changes are ready for review again.
cc. @c3024

**Comment by ChavdaSachin** [comment]
@c3024 could you put this PR on priority please. 
Since this PR is very large it keeps causing merge conflicts.
Thank you.

**Review by c3024 — APPROVED** [review]
### Screenshots/Videos
<details>
<summary>Android: Native</summary>




https://github.com/user-attachments/assets/650ff501-2832-4b2f-ad8e-dd39b979991c









</details>

<details>
<summary>Android: mWeb Chrome</summary>







https://github.com/user-attachments/assets/6bb0aeb7-31a6-41e4-855e-e0ef27e83c77







</details>

<details>
<summary>iOS: Native</summary>





https://github.com/user-attachments/assets/5697b7a2-bbf9-4903-85ed-6ecc6c543974










</details>

<details>
<summary>iOS: mWeb Safari</summary>









https://github.com/user-attachments/assets/4c4a78e1-adaa-4117-9943-8942e73b7b53







</details>

<details>
<summary>MacOS: Chrome / Safari</summary>






https://github.com/user-attachments/assets/e109aa4f-19bb-4ee9-88de-70e5d27fa0e8










</details>

<details>
<summary>MacOS: Desktop</summary>








https://github.com/user-attachments/assets/7dbdb031-fbf7-4fa4-bdaf-1b3cff7bc42e








</details>

LGTM!

**Review by Julesssss — APPROVED** [review]
Nice work

**Comment by Julesssss** [comment]
We can make some of these test cases new regression tests once the PR is deployed.

**Comment by IuliiaHerets** [comment]
This PR is failing for Android because of issue #53812

File: src/pages/workspace/expensifyCard/issueNew/IssueNewCardPage.tsx
Comment (allroundexperts) [inline src/pages/workspace/expensifyCard/issueNew/IssueNewCardPage.tsx:74]: We should have added a back button press functionality here. Not doing so caused https://github.com/Expensify/App/issues/55720

File: src/pages/workspace/companyCards/WorkspaceCompanyCardsPage.tsx
Comment (rojiphil) [inline src/pages/workspace/companyCards/WorkspaceCompanyCardsPage.tsx:75]: We decided to allow copilots to assign company cards in #85648

**Pull request merged**
